### PR TITLE
Adding tpi

### PR DESCRIPTION
**Pull request by casperdcl**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/main/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by shcheklein** [comment]
I confirm!

**Comment by casperdcl** [comment]
@conda-forge/help-python ready for review :)

File: recipes/tpi/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    - tpi = tpi.main:main
+
+requirements:
+  host:
+    - python >=3.4
+    - pip
+    - setuptools_scm >=3.4
+    - toml
+  run:
+    - python >=3.4
```
Comment (dopplershift) [inline recipes/tpi/meta.yaml:26]: Upstream lists a bunch more packages that are [required for install](https://github.com/iterative/tpi/blob/cc2683e8f4a45bb451f574c860df3b651bfd1c9a/setup.cfg#L62-L67). Any reason those are not listed, but instead only the optional `sshfs` is listed?

Comment (casperdcl) [reply recipes/tpi/meta.yaml:26]: quite right - just added them.

File: recipes/tpi/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    - funcy
+    - jinja2 >=2.0.0
+    - miutil >=0.7.0
+    - requests
+    - python-terraform >=0.10.1
+    - setuptools
+
+test:
+  imports:
+    - tpi
```
Comment (dopplershift) [inline recipes/tpi/meta.yaml:37]: Maybe add some more imports to double check the dependencies? It doesn't seem like just `import tpi` really exercises the code (`__init__.py` doesn't seem to do too much).
```suggestion
    - tpi
    - tpi.bin
    - tpi.terraform
```

**Comment by dopplershift** [comment]
`miutil` hasn't yet been packaged for conda-forge, so you'll need to add a recipe for that as well.

**Comment by casperdcl** [comment]
PR here: https://github.com/conda-forge/staged-recipes/pull/16115

**Comment by casperdcl** [comment]
@dopplershift now that #16115 is merged, can the tests be re-run now, or do we have to wait for https://github.com/conda-forge/miutil-feedstock to be generated first?

**Comment by isuruf** [comment]
We have to wait for the conda package to be generated.

**Comment by casperdcl** [comment]
@dopplershift / @isuruf any reason why this is failing? https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=375530&view=logs&j=e35d4f76-8ff2-5536-d795-df91e63eb9f7&t=fa7b4b17-b6ff-5c9c-8cfc-15f888c92310&l=1121

**Comment by isuruf** [comment]
CDN sync issues. Restarted

File: recipes/tpi/meta.yaml (hunk 2 above)
Comment (dopplershift) [reply recipes/tpi/meta.yaml:37]: Was there a reason this was marked resolved without any changes?

Comment (casperdcl) [reply recipes/tpi/meta.yaml:37]: I'd pushed exctly these changes but somehow lost them in the rebase, srry. Thanks for spotting it!

**Review by dopplershift — APPROVED** [review]
Pending CI.

**Pull request merged**